jobs/bayesian_pipeline.py

```diff
@@ -1,218 +1,242 @@
-
-        # model initial generation where:
-        # train_x is the parameters we are optimizing
-        # train_y is the target that we are optimizing that is minimized or maximized
-        self.train_x, self.train_y = self._generate_initial_data(
-            self.system, self.initial_samples
-        )
-        self.scaled_train_x, self.standardized_train_y = self._scale_and_standardize(
-            self.train_x, self.train_y
-        )
-        self.mll, self.gp_model = self._initialize_model(
-            self.scaled_train_x, self.standardized_train_y
-        )
-
-    def run(self) -> None:
-        self.log.debug("Starting Bayesian Optimization")
-        for i in range(self.num_iterations):
-            new_x, new_y = self._optimize_acquisition_function(
-                self.acquisition_function, self.gp_model, self.train_y
-            )
-            self.train_x = torch.cat([self.train_x, new_x])
-            self.train_y = torch.cat([self.train_y.squeeze(), new_y])
-            self.gp_model.set_train_data(
-                inputs=self.train_x, targets=self.train_y, strict=False
-            )
-            fit_gpytorch_model(self.mll)
-            self._update_best_designs(new_x, new_y)
-            self.log.debug(f"Iteration {i+1}/{self.num_iterations} complete")
-        self._print_best_designs()
-        self._find_analytical_results()
+    def __init__(self, conf: dict) -> None:
+        self.config: dict = conf
+        self.bayesian_setting: dict = self.config["job"]["BayesianOptimization"]
+        self.cf: EndureCost = EndureCost(self.bayesian_setting["max_levels"])
+        self.log: logging.Logger = logging.getLogger(self.config["log"]["name"])
+
+        self.system: System = System(**self.bayesian_setting["system"])
+        self.workload: Workload = Workload(**self.bayesian_setting["workload"])
+        self.h_bounds: torch.Tensor = torch.tensor([self.bayesian_setting["bounds"]["h_min"],
+                                                    self.bayesian_setting["system"]["H"]])
+        self.T_bounds: torch.Tensor = torch.tensor([self.bayesian_setting["bounds"]["T_min"],
+                                                    self.bayesian_setting["bounds"]["T_max"]])
+        self.policy_bounds: torch.Tensor = torch.tensor([0.0, 1.0])
+        self.bounds: torch.Tensor = torch.stack([self.h_bounds, self.T_bounds, self.policy_bounds], dim=-1)
+        self.initial_samples: int = self.bayesian_setting["initial_samples"]
+        self.acquisition_function: str = self.bayesian_setting["acquisition_function"]
+        self.q: int = self.bayesian_setting["batch_size"]
+        self.num_restarts: int = self.bayesian_setting["num_restarts"]
+        self.raw_samples: int = self.bayesian_setting["raw_samples"]
+        self.num_iterations: int = self.bayesian_setting["num_iterations"]
+        self.beta_value: float = self.bayesian_setting["beta_value"]
+        self.conn = initialize_database()
```

Comment:
Let's add a setting so we can set the name of the DB if we need to. Something simple in the TOML, maybe underneath the [io] section of the TOML

Reply:
I've set the db path and name now. It's in the Bayesian section for now since the db is writing only bo data but we could move it.

Reply:
Sounds good! That works too.